### PR TITLE
Implement date ranges on the Backend

### PR DESCRIPTION
**Pull request by Eochs**
Implements the graphql schema specified by @youngj in
trynmaps/metrics-mvp#301 and changes byDay to use an arbitrary array
of dates.
```
type TripMetrics {
    byDay(
        dateStrs: [String]
        startTime: String
        endTime: String
    ): [BasicIntervalMetrics]
}

type BasicIntervalMetrics {
    startDate: String
    endDate: String
    startTime: String
    endTime: String
    waitTimes: BasicWaitTimeStats
    tripTimes: BasicTripTimeStats
}

type BasicWaitTimeStats {
    median: Float
    percentile(percentile: Int!): Float
    probabilityLessThan(minutes: Int!): Float
}

type BasicTripTimeStats {
    median: Float
    percentile(percentile: Int!): Float
}
```

## Proposed changes

...

## Screenshot
image

...

## Link to demo, if any

...

### CODE REVIEW

**Comment by hathix** [comment]
Is this backwards-compatible with the old API? That is, will all old code still work, or will it have to be rewritten to handle the new API?

**Comment by hathix** [comment]
Also, can we record this API documentation somewhere? I think @youngj started API docs somewhere. We should make sure it updates in sync with the API code.

**Comment by Eochs** [comment]
> Is this backwards-compatible with the old API? That is, will all old code still work, or will it have to be rewritten to handle the new API?

Old code will still work, this adds an optional field (byDay) to TripMetrics, so old queries that don't include it will still work fine. Would be nice to have some tests though - I'll add some tests for the API in a separate PR. trynmaps/metrics-mvp#373

> Also, can we record this API documentation somewhere? I think @youngj started API docs somewhere. We should make sure it updates in sync with the API code.

Found it at [docs/metrics_api.md](https://github.com/trynmaps/metrics-mvp/blob/master/docs/metrics_api.md) I'll update it with the new changes.

**Comment by youngj** [comment]
@exxonvaldez is anyone working on the frontend blocked on this PR being merged? I think @eochs was working on adding API tests (which will be great to have), but tests could probably be added in a separate PR if anyone is blocked on this PR.

**Comment by exxonvaldez** [comment]
> @exxonvaldez is anyone working on the frontend blocked on this PR being merged? I think @Eochs was working on adding API tests (which will be great to have), but tests could probably be added in a separate PR if anyone is blocked on this PR.

I’m not sure, I had a cold last week and skipped hack night.  I took the date range UI issue, and haven’t started on that yet.  That isn’t blocked by this, I think, since the UI can send a set of dates for the current fetchGraphData query.  And then after that, somebody, probably me, would do an initial time series chart.

**Comment by hathix** [comment]
@Eochs what do you need to finish this PR? Are you blocked on anything?

File: docs/metrics_api.md [hunk 1]
```diff
@@ -637,12 +754,6 @@ Contains summary statistics for a single metric.
 | `percentiles` | [`[PercentileData]`](#percentiledata) | Data for percentile values of the computed metric. |
 | `histogram` | [`[HistogramBin]`](#histogrambin) | Data for rendering histograms on the frontend. |
 
-#### Parameters for `percentiles`
```
Comment (youngj) [inline docs/metrics_api.md]: I think this section shouldn't be deleted?

File: backend/models/schema.py [hunk 2]
```diff
@@ -7,6 +7,94 @@
...
+
+class BasicWaitTimeStats(ObjectType):
+    median = Float()
+    percentile = Float(percentile=Int(required=True))
+    probabilityLessThan = Float(minutes=Int(required=True))
+
+    def resolve_median(parent, info):
+        first_date = util.parse_date(parent["date_str"])
+        wait_time_median = wait_times.get_cached_wait_times(constants.AGENCY, first_date, "median", parent["start_time"], parent["end_time"])
+        return wait_time_median.get_value(parent["route_metrics"].route_id, parent["direction_id"], parent["start_stop_id"])
```
[the remark this answers is not in the document]
Comment (youngj) [reply backend/models/schema.py]: This issue still seems to be present

**Comment by hathix** [comment]
Going to merge this in and raise the things you mentioned as issues.

**Pull request merged**